satpy/tests/reader_tests/test_li_l2_nc.py

```diff
@@ -634,9 +632,6 @@ def test_coords_generation(self, filetype_infos):
             elevation_vals = elevation.values * point_height
             azimuth_vals *= -1
             lon_ref, lat_ref = projection(azimuth_vals, elevation_vals, inverse=True)
-            # Convert to float32:
-            lon_ref = lon_ref.astype(np.float32)
-            lat_ref = lat_ref.astype(np.float32)
```

Comment:
I'm a bit confused here: why do we need to remove the casting to float32 in the test reference data? What is the type of the `lon` and `lat` coming from the reader code, after this PR? I would have expected that after this PR the values should be in float32 from both sources and the test should pass without modifications...

Reply:
hmm ok, then could you please check where the `lon` and `lat` arrays become `float64` in the code? As requested, part of the fix we need is to avoid `float64` in favour of `float32` for float values...

Reply:
resolving this as we agreed to keep x/y as float64 as input to maintain the best precision when computing the lat/lon. The output lat/lon are now then casted to float32.